spec/00_nmatrix_spec.rb

```diff
@@ -638,4 +638,54 @@
       end
     end
   end
+
+  context "#repeat" do
+    before do
+      @sample_matrix = NMatrix.new([2, 2], [1, 2, 3, 4])
+    end
+
+    it "checks count argument" do
+      expect{@sample_matrix.repeat(1, 0)}.to raise_error(ArgumentError)
+      expect{@sample_matrix.repeat(-2, 0)}.to raise_error(ArgumentError)
+    end
+
+    it "returns repeated matrix" do
+      expect(@sample_matrix.repeat(2, 0)).to eq(NMatrix.new([4, 2], [1, 2, 3, 4, 1, 2, 3, 4]))
+      expect(@sample_matrix.repeat(2, 1)).to eq(NMatrix.new([2, 4], [1, 2, 1, 2, 3, 4, 3, 4]))
+    end
+  end
+
+  context "#meshgrid" do
+    before do
+      @x, @y = [1, 2, 3], [4, 5]
+      @two_dim = NMatrix.new([2, 2])
+      @one_dim = NMatrix.new([1, 1, 3])
+      @expected_result = [NMatrix.new([2, 3], [1, 2, 3, 1, 2, 3]), NMatrix.new([2, 3], [4, 4, 4, 5, 5, 5])]
+      @expected_for_ij = [NMatrix.new([3, 2], [1, 1, 2, 2, 3, 3]), NMatrix.new([3, 2], [4, 5, 4, 5, 4, 5])]
+      @expected_for_sparse = [NMatrix.new([1, 3], [1, 2, 3]), NMatrix.new([2, 1], [4, 5])]
+    end
+
+    it "checks arrs count" do
+      expect{NMatrix.meshgrid([@x])}.to raise_error(ArgumentError)
+      expect{NMatrix.meshgrid([])}.to raise_error(ArgumentError)
+    end
+
+    it "checks arrs for actual dimensions" do
+      expect{NMatrix.meshgrid(@x)}.to raise_error(ArgumentError)
+      expect{NMatrix.meshgrid([@one_dim, @one_dim])}.to_not raise_error(ArgumentError)
+      expect{NMatrix.meshgrid([@two_dim, @two_dim])}.to raise_error(ArgumentError)
+    end
+
+    it "returns new NMatrixes" do
+      expect(NMatrix.meshgrid([@x, @y])).to eq(@expected_result)
+    end
```

Comment:
Could you also test these functions with both `Array` and `NMatrix` for `@x` and `@y`? You can just create an array at the top and iterate over it.